bot/cogs/utils.py

```diff
@@ -174,6 +198,78 @@ def check(m: Message) -> bool:
             f"as I detected unauthorised use by {msg.author} (ID: {msg.author.id})."
         )
 
+    @command()
+    async def zen(self, ctx: Context, *, search_value: Optional[str] = None) -> None:
+        """
+        Show the Zen of Python.
+
+        Without any arguments, the full Zen will be produced.
+        If an integer is provided, the line with that index will be produced.
+        If a string is provided, the line which matches best will be produced.
+        """
+        if search_value is None:
+            embed = Embed(
+                colour=Colour.blurple(),
+                title="The Zen of Python, by Tim Peters",
+                description=ZEN_OF_PYTHON
+            )
+
+            await ctx.send(embed=embed)
+            return
+
+        zen_lines = ZEN_OF_PYTHON.splitlines()
+
+        # check if it's an integer. could be negative. why not.
+        is_negative_integer = search_value[0] == "-" and search_value[1:].isdigit()
+        if search_value.isdigit() or is_negative_integer:
+            index = int(search_value)
+
+            try:
+                line = zen_lines[index]
+            except IndexError:
+                embed = Embed(
```

Comment:
Raise a `BadArgument` with a message instead and let the error handler cog take care of the rest.

Reply:
This section has been tidied a little and now raises a BadArgument for the global error handler to deal with.